src/Service.Tests/Unittests/ConfigValidationUnitTests.cs

```diff
@@ -2017,6 +2017,72 @@ string[] linkingTargetFields
             }
         }
 
+        /// <summary>
+        /// This test checks that the final config used by runtime engine doesn't lose the directory information
+        /// if provided by the user.
+        /// It also validates that if config file is provided by the user, it will be used directly irrespective of
+        /// environment variable being set or not. 
+        /// When user doesn't provide a config file, we check if environment variable is set and if it is, we use
+        /// the config file specified by the environment variable, else we use the default config file.
```

Comment:
nit for future consideration. each of this bullet points would have been good to be separate tests for readability purposes. While there may be some duplicate code, the tests would be easier to read and have single purpose.